### ISSUE TITLE
Interoperability on the third-party libraries

### ISSUE THREAD

**Issue by FabioPinheiro** [roots-id/didcomm-mediator#21]
Currently, this mediator needs to have the hard-coded bytes inside of the encoded message.
`"typ" : "application/didcomm-plain+json"`

`typ` is not part of the specification.

This requirement comes from the Did Comm library implementation https://github.com/sicpa-dlab/didcomm-python/pull/93

**Comment by rodolfomiranda** [comment]
`typ` is the IANA media type. It's described on the spec [here](https://identity.foundation/didcomm-messaging/spec/v2.0/#iana-media-types). It reads:
"this spec recommends the JOSE convention of [using typ to make JOSE structure formats self-describing](https://tools.ietf.org/html/rfc7515#section-4.1.9). This is particularly helpful in the outermost envelope of any DIDComm message, before unwrapping begins."
Probable all sicpa libraries are implemented with that recommendation since we tested with the JVM and Rust and have not seen that error.

**Comment by FabioPinheiro** [comment]
On the outermost envelope, I have `"typ":"application/didcomm-encrypted+json"`

```
{
"ciphertext":"...",
"protected":"eyJlcGsiOnsia3R5IjoiT0tQIiwiY3J2IjoiWDI1NTE5IiwieCI6IndtSU90Sm03MVUwTGI3NDJqYThudFo1NjlGWDlqRjlBVVJ0cm1WYTluU28ifSwiYXB2IjoiLWNOQ3l0eFVrSHpSRE5SckV2Vm05S0VmZzhZcUtQVnVVcVg1a0VLbU9yMCIsInR5cCI6ImFwcGxpY2F0aW9uL2RpZGNvbW0tZW5jcnlwdGVkK2pzb24iLCJlbmMiOiJBMjU2Q0JDLUhTNTEyIiwiYWxnIjoiRUNESC1FUytBMjU2S1cifQ",
"recipients":[{...}],
"tag":"...",
"iv":"..."
}
```

```
{"epk":{"kty":"OKP","crv":"X25519","x":"wmIOtJm71U0Lb742ja8ntZ569FX9jF9AURtrmVa9nSo"},"apv":"-cNCytxUkHzRDNRrEvVm9KEfg8YqKPVuUqX5kEKmOr0","typ":"application/didcomm-encrypted+json","enc":"A256CBC-HS512","alg":"ECDH-ES+A256KW"}
```


But for what I understood from the code of `sicpa-dlab/didcomm-python`. It requires that field to be hard coded on the content (from `ciphertext`).

**Comment by rodolfomiranda** [comment]
I see. And they are actually harcoding it `d["typ"] = DIDCommMessageTypes.PLAINTEXT.value`

**Comment by FabioPinheiro** [comment]
There is already a fix. (just need to be released)

**Comment by FabioPinheiro** [comment]
There is still no new release. I tried to open an issue asking for it
https://github.com/sicpa-dlab/didcomm-python/issues/102

**Comment by FabioPinheiro** [comment]
I think this was fixed on version "0.3.2".
Let's update the [requirements.txt](https://github.com/roots-id/didcomm-mediator/blob/main/requirements.txt) to mention that version and close this.
